wpimath/src/main/java/edu/wpi/first/math/geometry/BoundVector2d.java

```diff
@@ -0,0 +1,191 @@
+package edu.wpi.first.math.geometry;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import edu.wpi.first.math.geometry.struct.BoundVector2dStruct;
+import edu.wpi.first.util.struct.StructSerializable;
+import java.util.Objects;
+
+/** Represents a bound vector in a vector field. Has an origin, and x/y component vectors. */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonAutoDetect(getterVisibility = JsonAutoDetect.Visibility.NONE)
+public class BoundVector2d implements StructSerializable {
+  private final Translation2d m_position;
+  private final Translation2d m_components;
+
+  /**
+   * Constructs a vector with the specified position and component vectors.
+   *
+   * @param position The position of the vector.
+   * @param components The X/Y component vectors.
```

Comment:
What do you mean by "component vectors" here? This variable is poorly named. The two fields in a Translation2d are scalars, not vectors.

Reply:
A scalar only has a magnitude while a vector has both direction and magnitude right? The direction can be just the sign of a number, and isn't necessarily an angle. `Translation2d.getX()` and `Translation2d.getY()` would be vectors in their respective axes. This needs to be made more clear though.

Reply:
If we have a vector <x, y>, we don't consider the elements x and y to also be vectors. If we did, what we actually have is a matrix.